extension/llm/tokenizer/utils.py

```diff
@@ -4,16 +4,23 @@
 # This source code is licensed under the BSD-style license found in the
 # LICENSE file in the root directory of this source tree.
 
+from typing import Optional
+
 from executorch.examples.models.llama.tokenizer.tiktoken import Tokenizer as Tiktoken
 from executorch.extension.llm.tokenizer.tokenizer import (
     Tokenizer as SentencePieceTokenizer,
 )
 
 
-def get_tokenizer(tokenizer_path):
-    try:
-        tokenizer = SentencePieceTokenizer(model_path=str(tokenizer_path))
-    except Exception:
-        print("Using Tiktokenizer")
-        tokenizer = Tiktoken(model_path=str(tokenizer_path))
+def get_tokenizer(tokenizer_path: str, tokenizer_config_path: Optional[str] = None):
```

Comment:
I don't like how this is written lol. If we are taking both arguments it's ambiguous which one is the source of truth.

Reply:
A better way is to ask the user to pass in a directory. We then try our best to determine if the directory contains a `tokenizer.[model|json]` and try to parse it. If no such file or the format is wrong, we error out.

Reply:
Discussion continued in above comment ^